### PR TITLE
electronim: init at 0.0.102

### LINKED ISSUE #223811: Package request: electronim
**Project description**

Free/Libre open source Electron based multi instant messaging (IM) client.

Combine all your IM applications (or whatever you want) in a single browser (Electron) window.

**Metadata**

* homepage URL: https://github.com/manusa/electronim
* source URL: https://github.com/manusa/electronim
* license: Apache-2.0
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/223811

This PR adds 1 package: `electronim`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/electronim/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    owner = "manusa";
+    repo = "electronim";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-DNEhptYu9JV0jQw3q4BIEdPiIHejHei1r7gznvKL37I=";
+  };
+
+  patches = [ ./remove-electron-package-json.patch ];
+
+  postPatch = ''
+    cp ${./package-lock.json} package-lock.json
```
Comment (wegank) [inline pkgs/applications/networking/instant-messengers/electronim/default.nix:24]: Including a huge `package-lock.json` in nixpkgs is considered unacceptable by some committers. Could you please try upstreaming it?

Comment (TomaSajt) [reply pkgs/applications/networking/instant-messengers/electronim/default.nix:24]: I'll try

Comment (TomaSajt) [reply pkgs/applications/networking/instant-messengers/electronim/default.nix]: I came up with a better solution, not requiring a changed package-lock.json
I might also upstream this patch later

**Comment by TomaSajt** [comment]
Not really interested in maintaining this anymore. Anyone still interested is free to open another PR based on this PR.

**Pull request closed without merging**